controllers/managedfusionoffering_controller.go

```diff
@@ -85,11 +88,49 @@ func pluginReconcile(r *ManagedFusionOfferingReconciler) (ctrl.Result, error) {
 	if _, found := r.managedFusionOffering.Spec.Config["onboardingValidationKey"]; !found {
 		return ctrl.Result{}, fmt.Errorf("%s CR does not contain onboardingValidationKey entry", r.managedFusionOffering.Name)
 	}
+	if err := reconcileOnboardingValidationSecret(r); err != nil {
+		return ctrl.Result{}, err
+	}
 	return ctrl.Result{}, nil
 }
 
+func reconcileOnboardingValidationSecret(r *ManagedFusionOfferingReconciler) error {
+	r.Log.Info("Reconciling onboardingValidationKeySecret")
+	onboardingValidationKeySecret := &corev1.Secret{}
+	onboardingValidationKeySecret.Name = "onboarding-ticket-key"
+	onboardingValidationKeySecret.Namespace = r.managedFusionOffering.Namespace
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, onboardingValidationKeySecret, func() error {
+		if err := r.own(onboardingValidationKeySecret); err != nil {
+			return err
+		}
+		onboardingValidationData := fmt.Sprintf(
+			"-----BEGIN PUBLIC KEY-----\n%s\n-----END PUBLIC KEY-----",
+			strings.TrimSpace(r.managedFusionOffering.Spec.Config["onboardingValidationKey"]),
```

Comment:
I consider reaching into the unstructured config a bad practice. My preference will be to have a structured struct that we set up at the start of the reconciliation (with the validation) and use it throughout the reconciliation. 

This also brings the thought of using a marshaled YAML document as an offering config instead of a string-to-string map.

Reply:
> My preference will be to have a structured struct that we set up at the start of the reconciliation (with the validation) and use it throughout the reconciliation.

Updated

> This also brings the thought of using a marshaled YAML document as an offering config instead of a string-to-string map.

As discussed, we will see if this needs to be done in future.